src/adapter/debugAdapter.ts

```diff
@@ -262,6 +264,48 @@ export class DebugAdapter {
     return {};
   }
 
+  async _toggleSkipFileStatus(params: Dap.ToggleSkipFileStatusParams): Promise<Dap.ToggleSkipFileStatusResult> {
+    if (params.resource) {
+      const absPath = urlUtils.fileUrlToAbsolutePath(params.resource);
+      if (absPath) {
+        params.resource = absPath;
+      }
+    }
+
+    if (!this.sourceContainer.scriptSkipper) {
+      if (!this._thread || !this._thread.cdp()) {
+        return {};
+      }
+      this.sourceContainer.initializeScriptSkipper(new ScriptSkipper([]));
+      this.sourceContainer.scriptSkipper!.setBlackboxSender(this._thread.cdp().Debugger);
+    }
+
+    const dapSource = await this._isSourceInCurrentStack(params);
+    if (dapSource && params.resource) {
+      const source = this.sourceContainer.source(dapSource);
+      if (source) {
+        source._blackboxed = !source._blackboxed;
+      }
+      this.sourceContainer.scriptSkipper!.toggleSkipFileStatus(params.resource);
+      this._withThread(thread => thread.pause());
+    }
+    return {};
+  }
```

Comment:
Maybe call it `lookupSourceInCurrentStack` or something, "is" sounds like it would return a boolean